### PR TITLE
OtlpTracePipeline fixed to no longer set global provider

### PR DESCRIPTION
**Pull request by utpilla**
Towards #1592 

## Changes
- This PR modifies the OTLP trace pipeline to not setup the global meter provider itself.
- End users must use the returned provider (a clone of it, if they cannot give up ownership), and call `global::set_tracer_provider()` themselves, if they want to set a global meter provider.
- Updated documentation comments for logs

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: examples/tracing-jaeger/src/main.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ fn init_tracer() -> Result<opentelemetry_sdk::trace::Tracer, TraceError> {
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-    let _tracer = init_tracer().expect("Failed to initialize tracer.");
+    let _tracer_provider = init_tracer().expect("Failed to initialize tracer provider.");
```
Comment (cijothomas) [inline examples/tracing-jaeger/src/main.rs:32]: need to set global here, else the rest of the code won't work!

Comment (utpilla) [reply examples/tracing-jaeger/src/main.rs]: Updated.

File: examples/tracing-jaeger/src/main.rs [hunk 2]
```diff
@@ -29,7 +29,8 @@ fn init_tracer() -> Result<opentelemetry_sdk::trace::Tracer, TraceError> {
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-    let _tracer = init_tracer().expect("Failed to initialize tracer.");
+    let tracer_provider = init_tracer().expect("Failed to initialize tracer provider.");
+    global::set_tracer_provider(tracer_provider.clone());
```
Comment (cijothomas) [inline examples/tracing-jaeger/src/main.rs:33]: the code at the end of this file uses the global::shutdown_tracer_provider. Lets fix that as well.
i.e modify the example to call shutdown on the above tracer_provider, but don't remove the existing global::shutdown_tracer_provider public api for now. (that need separate discussion)

Comment (cijothomas) [reply examples/tracing-jaeger/src/main.rs:33]: and ofcourse, do this for all the examples! 
Ideally - please manually validate the examples still work by verifying traces in collector!

File: opentelemetry-otlp/CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ Instead of using
 `.with_config(Config::default().with_resource(RESOURCE::default()))` users must
 now use `.with_resource(RESOURCE::default())` to configure Resource when using
 `OtlpLogPipeline`.
+- The methods `OtlpTracePipeline::install_simple()` and `OtlpTracePipeline::install_batch()` now return `TracerProvider` instead of `Tracer`.
```
Comment (cijothomas) [inline opentelemetry-otlp/CHANGELOG.md:15]: also callout global is not being set, and it is user responsibility to do so.
its best to show before and after code the user must do to ensure same behavior as before.

Comment (lalitb) [reply opentelemetry-otlp/CHANGELOG.md:15]: Also good to mention this as breaking change.

Comment (utpilla) [reply opentelemetry-otlp/CHANGELOG.md]: Updated

**Review by cijothomas — APPROVED** [review]

File: examples/tracing-jaeger/src/main.rs [hunk 4]
```diff
@@ -10,7 +10,7 @@ use opentelemetry_semantic_conventions::resource::SERVICE_NAME;
 
 use std::error::Error;
 
-fn init_tracer() -> Result<opentelemetry_sdk::trace::Tracer, TraceError> {
+fn init_tracer() -> Result<opentelemetry_sdk::trace::TracerProvider, TraceError> {
```
Comment (TommyCpp) [inline examples/tracing-jaeger/src/main.rs:13]: Shall we rename this to `init_tracer_provider`?

**Review by TommyCpp — APPROVED** [review]
Good to merge once we rename all `init_tracer` to `init_tracer_provider`. Thanks a lot!

**Comment by utpilla** [comment]
Thanks for the review! I have addressed all the comments.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**